### ISSUE TITLE
Document or wiki

### ISSUE THREAD

**Issue by thomas-tri-tran** [transmute-industries/sidetree.js#109]
Do you have any document or wiki how to use this ?

**Comment by OR13** [comment]
@thomas-tri-tran we have lots of tests, they are the best source of documentation we have so far.

if you have a specific question, I am happy to point you to the relevant tests showing how to do it.

**Comment by OR13** [comment]
You may also find the spec version we support helpful: https://web.archive.org/web/20200721150053/https://identity.foundation/sidetree/spec/v0.1.0/

**Comment by thomas-tri-tran** [comment]
Actually, I am developing the marketplace app for trader to buy and sell artworks. The trader need to provide some identification document e.g the driver's license or passport ID  and some public profile like name, artwork category ,etc. How does the trader create the did, the profile and register it to the ledger? In the https://github.com/decentralized-identity/element there are some code snippets to create did and read did as below. 

Do you have any example how to do similar using sidetreejs? Another question is 

`const mks = new MnemonicKeySystem(MnemonicKeySystem.generateMnemonic());
const primaryKey = await mks.getKeyForPurpose("primary", 0);
const recoveryKey = await mks.getKeyForPurpose("recovery", 0);
const didDocumentModel = element.op.getDidDocumentModel(
  primaryKey.publicKey,
  recoveryKey.publicKey
);

// Generate Sidetree Create payload
const createPayload = element.op.getCreatePayload(didDocumentModel, primaryKey);

// Create the Sidetree transaction.
// This can potentially take a few minutes if you're not on a local network
const createTransaction = await element.batchScheduler.writeNow(createPayload);
const didUniqueSuffix = element.func.getDidUniqueSuffix(createPayload);
const did = `did:elem:ropsten:${didUniqueSuffix}`;
console.log(`${did} was successfully created`);`

**Comment by OR13** [comment]
generating sidetree operations. / did documents from mnemonics: https://github.com/transmute-industries/sidetree.js/blob/master/packages/wallet/src/__test__/generate-svip-operation-fixture.test.ts

generating keypairs from mnemonics:

https://github.com/transmute-industries/sidetree.js/blob/master/packages/wallet/src/__test__/generate-keypair-fixture.test.ts

This code is used here: http://staging.element.transmute.industries/

**Comment by OR13** [comment]
There is also the core crypto package, which might do what you want: https://github.com/transmute-industries/sidetree.js/tree/master/packages/crypto/src

**Comment by NickDarvey** [comment]
>This code is used here: http://staging.element.transmute.industries/

I don't suppose the source code for that workbench is laying around somewhere? I was running into issues getting it going in a browser, likely alleviated with the right shim or webpack configuration I'm sure...

**Comment by OR13** [comment]
@NickDarvey the source code for the workbenches is not public currently, but its built on:

- create react app
- https://github.com/w3c-ccg/universal-wallet-interop-spec/tree/master/packages/universal-wallet
- and this repo

we refactored sidetree crypto so the parts necessary to create a did can run in the browser.

If you want trying to reproduce the same functionality, I suggest you start with create react app and incrementally add the wallet and crypto dependencies... happy to review any challenges you face along the way, please open issues :)

**Comment by OR13** [comment]
I am fixing some version drift issues right now, so its possible they are related to what you are describing.

**Comment by NickDarvey** [comment]
No worries — I'll open a new issue with specifics if I run into troubles with the undrifted versions

**Comment by thomas-tri-tran** [comment]
@OR13 thanks for the information. The http://staging.element.transmute.industries/ is very help.
I try to create a new identity and convert to did Document. I found in the source code https://github.com/transmute-industries/sidetree.js/tree/master/packages/wallet/src/functions toDidDoc and toDidDocForProfile. What's the different between those two? What is the SVIP and why only this profile required?

**Comment by OR13** [comment]
The SVIP profile is for Ed25519 and X25519 which are allowed crypto for DHS SVIP Blockchain cohort. 

I recommend avoiding secp256k1 as much as possible :)

**Comment by thomas-tri-tran** [comment]
In the staging demo, why the public key generated type as JsonWebKey2020 instead of Ed25519VerificationKey2018 and X25519KeyAgreementKey2019. The id  did:did:elem:ropsten: seems not correct. Can I generate the did document without commit to the ledger?
{
        "id": "#z6MkvnV8iQ9rAUevfJ46Fo4H3GVDe5L67S7NtKtnFrSgceJv",
        "type": "JsonWebKey2020",
        "controller": "did:did:elem:ropsten:EiBijX8D1qo_UBhjpyAwNDpJujCOUBczQqJ2n5ewTkLgqw",
        "publicKeyJwk": {
          "crv": "Ed25519",
          "x": "8qfTQZrkSPYFlL9c2YDqvD_ThTGP511HcY3A3PsYyAs",
          "kty": "OKP",
          "kid": "B2Rd_JM5PjsqcVs87CwA-GXhRYSa-3T47IrYvpYa734"
        }
      }

**Comment by OR13** [comment]
https://github.com/w3c-ccg/lds-jws2020

TL;DR; JsonWebKey2020 supports most popular JWK key types...

its also supported by DID Core https://w3c.github.io/did-spec-registries/#jsonwebkey2020

You can generate a DID without ledger commitment... just use:

https://github.com/transmute-industries/sidetree.js/blob/master/packages/wallet/src/functions/toDidDocForProfile.test.ts

This will generate a DID Document with a "Long Form DID".

But beware that Long Form DIDs are not stable / safe for long term use.

**Comment by thomas-tran** [comment]
Do you have the source code for this http://staging.element.transmute.industries/  available publicly?

Also, if I got the kyc , how can I associate it with the sidetree did?

**Comment by OR13** [comment]
@thomas-tran you are correct that the source for that URL is not public currently.

That version of element is on the ethereum ropsten testnet, and not suitable for production currently.

**Comment by kc1116** [comment]
Does the element api and element app get deprecated ?

**Comment by kc1116** [comment]
How do you deploy a "element sidetree node" ?Is this repo only the javascript lib now ?

**Comment by OR13** [comment]
@kc1116 

This repo is a mono repo for sidetree based did methods we support. (we currently don't publish docker containers or ui / app code for working with those methods).

http://staging.element.transmute.industries/ (closed source -- built from this repo)

http://element-did.com/ (deprecated / do not use).

**Comment by OR13** [comment]
@kc1116 worth noting this repo is pre sidetree v1... we are tired of chasing Microsoft pushing breaking changes and are waiting for them to hit GA / get fully committed to ION, before we attempt to catch up to sidetree spec v1... but this repo is used in our products and meets our needs for now.

**Comment by kc1116** [comment]
Okay so you would recommend for us to build our own UI, deprecate usage of the old element repo, and use this repo as the replacement element-lib ?

**Comment by OR13** [comment]
@kc1116 yes, I would avoid relying on the old element-lib repo, we won't continue to support it.

**Comment by kc1116** [comment]
@OR13 Is there any other type of documentation anywhere. How should one go out using this Repo. For example installing it to use in another project.

**Comment by OR13** [comment]
@kc1116 its a great question, we don't have good examples of that really... It sounds like what we need is a small PoC / standalone example of an api and a front end that uses this repo's packages.

We have some react / storybook demo's of wallet / edv integration here: https://github.com/w3c-ccg/universal-wallet-interop-spec/tree/master/packages

I think we can probably standup something similar for this repo with a few days worth of work.

**Comment by thomas-tran** [comment]
It would be great if you could provide the small POC or example how to use the API as I am struggling to search for the example in the test cases especially how to create the did , update and revoke it.

**Comment by OR13** [comment]
thanks for this feedback!

I can't promise you a timeline for when we will have this done, but the next best thing is to look at the tests:

https://github.com/transmute-industries/sidetree.js/blob/main/packages/did-method-element/src/__test__/sidetree-wallet-svip-profile-conformance.test.ts

^ for example, these tests show how to create and recover using the wallet code in this repo and a locally running api.

**Comment by kc1116** [comment]
It will get us super far if you can reply with atleast documentation on how to initialize this [element class](https://github.com/transmute-industries/sidetree.js/blob/c8e893a848cc57664b09e91771e7f0068c104bbb/packages/did-method-element/src/Element.ts#L17) . The test above are a perfect example, documentation is badly needed though.

**Comment by OR13** [comment]
@kc1116  https://github.com/transmute-industries/sidetree.js/blob/c8e893a848cc57664b09e91771e7f0068c104bbb/packages/did-method-element/src/Element.spec.ts#L40

**Comment by OR13** [comment]
> documentation is badly needed though.

yes, sorry... and beware that we are still behind sidetree core / SIP-1... we are waiting for ION to be in GA before upgrading...

**Comment by kc1116** [comment]
Thanks for all the help, last question what are the current contract addresses for element ?

**Comment by OR13** [comment]
@kc1116 feel free to ask as many questions as you like :)

This is the contract address which is behind did:elem:ropsten:

https://ropsten.etherscan.io/address/0x920b7DEeD5CdE055260cdDBD70C000Bbd5b30997

and is powering https://staging.element.transmute.industries/

**Comment by BenjaminMoe** [comment]
Since it looked like the original issue was resolved, I'm going to go ahead and close this. We've been adding a lot of documentation in the form of markdown files, and trying to link them in the packages they are used in to make them visible as possible. If we find that we have a lot of mark down files that aren't code, we may switch over and start using the Wiki functionality.
